boa/src/syntax/parser/function/mod.rs

```diff
@@ -84,6 +91,14 @@ where
                 _ => FormalParameter::new(self.allow_yield, self.allow_await).parse(cursor)?,
             };
 
+            if self.unique && params.iter().any(|v| v.name() == next_param.name()) {
+                // https://tc39.es/ecma262/#prod-UniqueFormalParameters
+                return Err(ParseError::lex(LexError::Syntax(
+                    "Invalid duplicate formal parameter name".into(),
+                    pos,
+                )));
+            }
```

Comment:
I don't think this should be a lexing error. It's a normal parsing error, the lexing was correct, but in this context, it's a syntax error, so a parsing error.